pypsa/components.py

```diff
@@ -576,11 +633,15 @@ def set_snapshots(
             pnl = self.pnl(component)
             attrs = self.components[component]["attrs"]
 
-            for k, default in attrs.default[attrs.varying].items():
+            for k in pnl.keys():
                 if pnl[k].empty:  # avoid expensive reindex operation
                     pnl[k].index = self._snapshots
+                elif k in attrs.default[attrs.varying]:
+                    pnl[k] = pnl[k].reindex(
+                        self._snapshots, fill_value=attrs.default[attrs.varying][k]
+                    )
```

Comment:
this change is purely stylistic or?

Reply:
No, previously new snapshots have not been applied to custom attributes, so non-default attributes

Reply:
:ok_hand: